### DISCUSSION TITLE
Is longhorn taking too much resource

### DISCUSSION THREAD

**Discussion by liyimeng** [longhorn/longhorn#5095] [Q&A]
Longhorn doc suggest reserve so much CPU for longhorn on the node

```
Guaranteed Replica Manager CPU = The estimated max Longhorn volume replica count on a node * 0.1 / The total allocatable CPUs on the node * 100.
```
In my case, I have a 32 cores server with estimated 120 Vols will schedule into the node, the number will be come

120 * 0.1 /32 * 100 = 37.5 

with two managers, I have to reserve 75% of my server CPU power for longhorn,  little is left for real traffic, what I have done wrong?

**Comment by joshimoo** [comment]
You need roughly 10% of a cpu core per active replica/engine process.
So if you use the dedicated storage node architecture, (couple of dedicated nodes with high cpu/storage) then that node will primarily be providing it's cpu to the storage system in this case longhorn.

**Comment by liyimeng** [comment]
@joshimoo  `You need roughly 10% of a cpu core per active replica/engine process.`  This seem a pretty high number. are we leveraging technology zero-copy (https://en.wikipedia.org/wiki/Zero-copy) to minimise the load?
